many: remove dm-verity support from snap pack

sespiros opened the pull request:
Under the new design, generating dm-verity data via `snap pack` is not needed as support for integrity data was simplified and there is no extra logic or separate header anymore.

Jira: https://warthogs.atlassian.net/browse/FR-9880

alfonsosanchezbeato reviewed (commented):
Shouldn't also the files in `snap/integrity` be removed?

sespiros commented:
> Shouldn't also the files in `snap/integrity` be removed?

I didn't remove it in this one since that's the path I still use for the integrity API (see upcoming PRs such as https://github.com/canonical/snapd/pull/14872 which changes it almost completely). The snap integrity API (under `snap/integrity`) will be called by other various parts of snapd (overlord during snap installation, snap-bootstrap etc.).

pedronis commented on snap/pack/pack.go:
```diff
@@ -283,13 +280,6 @@ func mksquashfs(sourceDir, fName, snapType string, opts *Options) error {
 	}
 
-	if opts.Integrity {
-		err := integrity.GenerateAndAppend(fName)
```
shouldn't at least GenerateAndAppend also go away? or is it modified in a follow up?

sespiros replied:
It will go away, see https://github.com/canonical/snapd/pull/14872/files#diff-f2b95ee6cb86d0d94e54d356707f39c2e6014beef108da27d124daf04d6eb5daL115

Instead of removing everything under `snap/integrity` in this one and reintroduce it in the follow-up, I left it as is but I drastically change it in the follow-ups.

pedronis reviewed (commented):
looks good, but one question

pedronis reviewed (approved):
thanks

alfonsosanchezbeato reviewed (approved):
Thank you

pedronis commented:
@sespiros this spread test need changing or removing:

google:ubuntu-24.10-64:tests/main/snap-pack-integrity

sespiros commented:
@pedronis `--append-integrity-data` doesn't exist anymore so I removed the test. 

@Meulengracht gh UI is confusing, I assume you merged master in my branch right? Somehow this was forced pushed in my tree, not sure how that happened, I thought only myself could write my tree :D Anyway what I did was fetch/merge my branch, added the new commit on top and pushed my branch. I didn't use `--force` so I assume it did the right thing.

Meulengracht commented:
> @pedronis `--append-integrity-data` doesn't exist anymore so I removed the test.
> 
> @Meulengracht gh UI is confusing, I assume you merged master in my branch right? Somehow this was forced pushed in my tree, not sure how that happened, I thought only myself could write my tree :D Anyway what I did was fetch/merge my branch, added the new commit on top and pushed my branch. I didn't use `--force` so I assume it did the right thing.

Correct, i updated your branch due to test changes, so I want them rerun with newest master, same for your other PR

olivercalder commented:
This appears to have broken formatting in `cmd/snap/cmd_pack.go`, not sure how the static checks passed.
